### PR TITLE
Make TadGAN Build Once

### LINKED ISSUE #261: Adjusting builds for TadGAN
* Orion version: 0.2.0
* Python version: 3.7
* Operating System: macOS 10.15.6

### Description

In the current version of TadGAN, we build the model with every fit call
https://github.com/sintel-dev/Orion/blob/6548161e240a413ba011a699d109eb8a1e50c148/orion/primitives/tadgan.py#L290

When we can to continue training, we do not necessarily need to rebuild the model. To fix this issue, we can introduce a flag to indicate if/when to build the model

```python3
if build:
    self._augment_hyperparameters(X, y, kwargs)
    self._set_shapes()
    self._build_tadgan(**kwargs)

self._fit(X, y)
```
---
### EDIT

Rather than using a `build` parameter, make "rebuilding" a model not possible. If the user wishes to "rebuild" a model then they can create a new Orion instance.

Changes to reflect this is

```python3
if not self._fitted:
    self._augment_hyperparameters(X, y, kwargs)
    self._set_shapes()
    self._build_tadgan(**kwargs)

self._fit(X, y)
self._fitted = True
```

### PR DESCRIPTION
**Pull request by sarahmish**
Resolve #261

### CODE REVIEW

**Review by frances-h — APPROVED** [review]
LGTM! 👍

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.6, 3.7]
+        python-version: [3.7]
```
Comment (dyuliu) [inline .github/workflows/tests.yml:17]: Do you think it is helpful to highlight in the readme file that currently Orion is developed and fully tested in python 3.7?

File: orion/primitives/tadgan.py [hunk 2]
```diff
@@ -273,7 +274,7 @@ def _fit(self, X, target):
             print('Epoch: {}/{}, [Dx loss: {}] [Dz loss: {}] [G loss: {}]'.format(
                 epoch, self.epochs, cx_loss, cz_loss, g_loss))
 
-    def fit(self, X, y=None, **kwargs):
+    def fit(self, X, y=None, build=False, **kwargs):
         """Fit the TadGAN.
 
         Args:
```
Comment (dyuliu) [inline orion/primitives/tadgan.py:280]: Update the comment here to include `build`?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (sarahmish) [reply .github/workflows/tests.yml:17]: I will do that in another PR when we update the readme

**Review by dyuliu — APPROVED** [review]

**Pull request merged**